### PR TITLE
Added from/to height query filters to the transaction search endpoints

### PR DESCRIPTION
**Pull request by Vektrat**
Closes https://github.com/nemtech/catapult-rest/issues/469

### CODE REVIEW

**Comment by Vektrat** [comment]
note that this aims to allow the clients to be able to filter transactions by height (period) range which is what the architecture offers

as you know, the chain does not contain explicit real world timestamp data, thus if user experience needs to be improved, it is in the hands of the SDK/clients to offer a way to convert height to timestamp (probably involving timestamp fields and nemesis block creation data)

File: spec/parameters/query/fromHeight.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name: fromHeight
+in: query
+description: Filter from block height.
```
Comment (segfaultxavi) [inline spec/parameters/query/fromHeight.yml:3]: The description should say if the block with that exact same height is included or not (inclusive or exclusive filter).

Also, the word "filter" can be a bit ambiguous. I would clarify it adding something like `Only blocks with height greater than this one are returned.`.

File: spec/parameters/query/toHeight.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: toHeight
+in: query
+description: Filter up to block height.
```
Comment (segfaultxavi) [inline spec/parameters/query/toHeight.yml:3]: Same comment as above.

**Review by segfaultxavi — CHANGES_REQUESTED** [review]

**Review by martinayora — APPROVED** [review]

**Review by martinayora — APPROVED** [review]

**Review by segfaultxavi — APPROVED** [review]
My concerns have been addressed.

**Comment by Vektrat** [comment]
this is ready but waiting for fernando to give green light (he asked us to wait to merge this because his schemas would be regenerated or something)

**Pull request merged**